androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/EnhancedVideoPlayer.kt

```diff
@@ -41,11 +45,38 @@ fun EnhancedVideoPlayer(
         }
     }
 
+    Player(
+        exoPlayer = exoPlayer,
+        useControls = useControls,
+        fullScreen = fullScreen,
+        controlsDecoration = controlsDecoration
+    )
+}
+
+@androidx.annotation.OptIn(UnstableApi::class)
+@Composable
+fun Player(
+    exoPlayer: ExoPlayer,
+    controlsDecoration: ControlsDecoration,
+    useControls: Boolean = false,
+    fullScreen: Boolean = true
+) {
+    var isPlaying by remember { mutableStateOf(exoPlayer.isPlaying) }
```

Comment:
nit: isn't this a `val`?

Reply:
This makes sense to be on `EnhancedVideoPlayer` composable, as we are already instantiating and adding listeners to `exoPlayer` there. Then, this component can also wrap into an `AndroidView`, adding custom controls if necessary. This class can even be removed as it is redundant with `EnhancedVideoPlayer` composable

Reply:
It could be a val, but then I'd have to change it to:

```kotlin
val mutableState = remember
```

And to update the value I'd have to do:

```kotlin
 `mutableState.value = newState`. 
```

I don't have any preferences on this, to be honest, the [doc](https://developer.android.com/jetpack/compose/state#state-in-composables) says this is exactly the same thing, but in the QS we were using this pattern, so I just followed it